### PR TITLE
Pj adding LR model

### PR DESCRIPTION
**Pull request by PeterJackNaylor**
Heya!

So i implemented the logistic regression and it was working.

So I was a bit dumbed and I implemented it without doing a pull before.
To check it was working I simply run ```python two_backbones.py```. With the changes in the latest pull, the testing thing doesn't work anymore. Would you mind debugging it to have a working version of the test for two_backbones? At first, it was missing the input ```glofas```, which I added similarly to era5l. but now its the ``` self.hydroatlas_class_mask = [
            (band.split("_")[1] in ["cl", "id"]) for band in self.hydroatlas_bands
        ]``` which is not working, and I don't really understand why you split it here, because in the test code of that script, self.hydroatlas_bands is simply a list of number, without strings.

Anyway, after this I should be able to run the logistic regression model without any problem. It has been under-optimized as this a local-network only model, hence I would need to modify the dataloader to not load the context input data anymore to fully optimize it :-) !

### CODE REVIEW

File: gff/models/two_backbones.py [hunk 1]
```diff
@@ -39,8 +38,8 @@ def get_empty_norms(era5_bands: int, era5l_bands: int, glofas_bands: int, hydroa
 
 
 options_metnet = {
-    "dim": 64,
-    "attn_depth": 12,
+    "dim": 32,
+    "attn_depth": 8,
```
Comment (Multihuntr) [inline gff/models/two_backbones.py:42]: Wait, why are you changing the metnet configuration?

Comment (PeterJackNaylor) [reply gff/models/two_backbones.py:42]: Ah yes, it was too large for my gpu at the time, i can remove this change

File: gff/models/two_backbones.py [hunk 2]
```diff
@@ -430,4 +444,7 @@ def forward(self, ex):
             out = model(ex)
             assert out.shape == (8, 3, 224, 224)
             print("🗸", end="")
+
+    print("Peter checking weight importance for LR")
+    lr_model.print_weights(model, feature_names=["S1 (0)", "S1 (1)", "Local DEM"])
```
Comment (Multihuntr) [inline gff/models/two_backbones.py:449]: This shouldn't be here. At the very least, it should be in an if-statement.

But, I don't want a print statement with your name in it in the final code base. :laughing:

**Comment by Multihuntr** [comment]
Leave the hydroatlas bands bug with me. I'll get to it after I've implemented something else.

File: gff/models/two_backbones.py [hunk 3]
```diff
@@ -229,6 +229,12 @@ def __init__(
                 op_type="2d",
                 cond_norm_affine=cond_norm_affine,
             )
+        elif backbone == "LR":
+            self.context_embed = None
+            self.local_embed = lr_model.LogisticRegression(n_channels=local_input_dim - context_embed_output_dim + 1, out_channels=n_predict)
+        elif backbone == "Context+LR":
+            self.context_embed = None # put your best model here
+            self.local_embed = lr_model.LogisticRegression(n_channels=local_input_dim, out_channels=n_predict, cond_dim=lead_time_dim)
```
Comment (Multihuntr) [inline gff/models/two_backbones.py:237]: cond_dim isn't used. Why is it included here?

**Comment by Multihuntr** [comment]
Also, no changes are necessary to data loader. Not using any context data sources is just a configuration change. Currently the two_backbones explicitly checks that you are using at least one context data source to make a prediction (see two_backbones.py:129 in this PR). 

My vote is to just bite the bullet and enable local-only training for all models. So, if using no context data sources, it replaces the context_embed with a dummy model that outputs `[B, 0, H, W]` as described earlier before. This results in very few code changes, and harmonises across different models. 

Opinion?

File: gff/models/two_backbones.py (hunk 2 above)
Comment (PeterJackNaylor) [reply gff/models/two_backbones.py:449]: bouuuuh :p

 fair enough ^^

**Pull request merged**